### PR TITLE
sci-physics/{oneloop,cuttools,openloops}: add {3.6, 1.8.2, 2.1.3}

### PR DESCRIPTION
**Pull request by APN-Pucky**
At least two imminent bigger problems here (before I clean it up further):

- kind_types.f90 is from openloops2, which I'd link to PR sooner or later, since they modified oneloop I'd include these changes here. Alternatively, openloops2 could just use their own version of oneloop, but that seems a bit wrong (unless they perform many more changes).
- I dropped the 2020.07.31, since I found a the latest official version 3.6 and therefore the 2020.07.31 version is just from the bitbucket and should be called 3.6_p2020.07.31, IMO. 
- Maybe I should just throw the openloops PR inside of here too? 

How does Gentoo typically handle these (rare science mess) cases?

### CODE REVIEW

**Comment by AndrewAmmerlaan** [comment]
> How does Gentoo typically handle these (rare science mess) cases?

Do I understand correctly that this bundles a modified version of a dependency? 

In this case I would say it depends on which modifications they made and how many. If the changes are generally useful I think we can patch the dependency in its own package. Otherwise I think the bundled dependency should be in SRC_URI.

File: sci-physics/oneloop/oneloop-3.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+SRC_URI="http://helac-phegas.web.cern.ch/helac-phegas/tar-files/${MY_P}.tgz"
+S="${WORKDIR}/${MY_P}"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="+dpkind +qpkind qpkind16 dpkind16 qdcpp ddcpp mpfun90 arprec tlevel cppintf kind_types"
+REQUIRED_USE="
+	?? ( dpkind dpkind16 ddcpp )
+	?? ( qpkind qpkind16 qdcpp )
```
[the remark this answers is not in the document]
Comment (APN-Pucky) [reply sci-physics/oneloop/oneloop-3.6.ebuild:26]: I don't think so one is quad and other double precision.

**Comment by APN-Pucky** [comment]
> > How does Gentoo typically handle these (rare science mess) cases?
> 
> Do I understand correctly that this bundles a modified version of a dependency?
> 
> In this case I would say it depends on which modifications they made and how many. If the changes are generally useful I think we can patch the dependency in its own package. Otherwise I think the bundled dependency should be in SRC_URI.

I guess I'll add the other stuff here too, so one can explicitly check that it works with the final package.

File: sci-physics/oneloop/oneloop-3.6.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndrewAmmerlaan) [reply sci-physics/oneloop/oneloop-3.6.ebuild:26]: :O 

I must have been very tired when I was looking at this, sorry.

File: sci-physics/cuttools/cuttools-1.8.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit toolchain-funcs
+
+DESCRIPTION="Computing 1-loop amplitudes at the integrand level"
+HOMEPAGE="https://www.ugr.es/~pittau/CutTools/"
+SRC_URI="http://www.ugr.es/~pittau/CutTools/${PN}_v${PV}.tar.gz"
```
Comment (AndrewAmmerlaan) [inline sci-physics/cuttools/cuttools-1.8.2.ebuild:10]:
```suggestion
SRC_URI="https://www.ugr.es/~pittau/CutTools/${PN}_v${PV}.tar.gz"
```

File: sci-physics/cuttools/cuttools-1.8.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	virtual/fortran
+"
+src_prepare() {
+	default
+	sed -i 's/^ALL =.*$/ALL = $(CTS)/' src/makefile ||  die
+	use dummy && cp "${FILESDIR}"/mpnumdummy.f90 src/cts/mpnumdummy.f90
+	use dummy && sed -i 's/CTS =/CTS = mpnumdummy.o/' src/makefile
```
Comment (AndrewAmmerlaan) [inline sci-physics/cuttools/cuttools-1.8.2.ebuild:30]: These should `|| die`, so we need a proper if statement here.

File: sci-physics/cuttools/cuttools-1.8.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+src_compile() {
+	emake -j1 FFLAGS="${FFLAGS} -I${ESYSROOT}/usr/include -fPIC -std=legacy"
+	tc-export AR CXX
+	cd includects
```
Comment (AndrewAmmerlaan) [inline sci-physics/cuttools/cuttools-1.8.2.ebuild:36]:
```suggestion
	cd includects || die
```

File: sci-physics/cuttools/cuttools-1.8.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	${AR} -x libcts.a
+	${CXX} -shared *.o -o lib${PN}.so
```
Comment (AndrewAmmerlaan) [inline sci-physics/cuttools/cuttools-1.8.2.ebuild:38]:
```suggestion
	${AR} -x libcts.a || die
	${CXX} ${CXXFLAGS} -shared *.o -o lib${PN}.so || die
```

File: sci-physics/cuttools/cuttools-1.8.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+src_install() {
+	cd includects
+	dolib.so lib${PN}.so
+	cd ..
+	mv includects ${PN}
+	rm ${PN}/*.a
+	rm ${PN}/*.so
+	rm ${PN}/*.o
+	doheader -r ${PN}
```
Comment (AndrewAmmerlaan) [inline sci-physics/cuttools/cuttools-1.8.2.ebuild:49]:
```suggestion
	cd includects || die
	dolib.so lib${PN}.so
	cd .. || die
	mv includects ${PN} || die
	rm ${PN}/*.a || die
	rm ${PN}/*.so || die
	rm ${PN}/*.o || die
	doheader -r ${PN}
```

File: sci-physics/cuttools/cuttools-1.8.2.ebuild [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+IUSE="+dummy"
+DEPEND="
+	sci-physics/qcdloop
+	sci-physics/oneloop[dpkind,qpkind16,-qpkind,-tlevel]
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	virtual/fortran
+"
```
Comment (AndrewAmmerlaan) [inline sci-physics/cuttools/cuttools-1.8.2.ebuild:25]: `fortran-2.eclass` :P

File: sci-physics/openloops/openloops-2.1.2.ebuild [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+DEPEND="${RDEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/${P}-python3.12.patch"
+	"${FILESDIR}/${P}-ldflags.patch"
+)
+
+src_prepare() {
+	default
+	mv openloops.cfg.tmpl openloops.cfg
```
Comment (AndrewAmmerlaan) [inline sci-physics/openloops/openloops-2.1.2.ebuild:39]:
```suggestion
	mv openloops.cfg.tmpl openloops.cfg || die
```

File: sci-physics/openloops/openloops-2.1.2.ebuild [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+link_flags = ${LDFLAGS} -I${ESYSROOT}/usr/include/ -I${ESYSROOT}/usr/include/cuttools -lcollier
+cc = $(tc-getCC)
+cxx = $(tc-getCXX)
+fortran_compiler = $(tc-getFC)
+process_src_dir = ${EPREFIX}/opt/${MY_P}/process_src/
+process_obj_dir = ${EPREFIX}/opt/${MY_P}/process_obj/
+process_lib_dir = ${EPREFIX}/opt/${MY_P}/proclib/
+release = $PV
+import_env = @all
+EOF
```
Comment (AndrewAmmerlaan) [inline sci-physics/openloops/openloops-2.1.2.ebuild:60]: Here you can also use `-EOF` instead so you can use indentation

File: sci-physics/cuttools/cuttools-1.8.2.ebuild (hunk 7 above)
Comment (APN-Pucky) [reply sci-physics/cuttools/cuttools-1.8.2.ebuild]: I am going to patch that now into my pkgcheck...

**Review by AndrewAmmerlaan — APPROVED** [review]

**Pull request closed without merging**